### PR TITLE
docs: fully revised lint configuration guide. First version

### PR DESCRIPTION
**Pull request by bandantonio**
## Check yourself

- [ ] Code is linted
- [ ] Tested with redoc/reference-docs/workflows
- [ ] All new/updated code is covered with tests

## Security

- [ ] Security impact of change has been considered
- [ ] Code follows company security practices and guidelines

### CODE REVIEW

File: docs/configuration/lint.md [hunk 1]
```diff
@@ -42,45 +27,190 @@ lint:
...
+The first match will win in the event when a URL matches multiple patterns. Therefore, only the header from the first match will be used in the request.
+
+### Preprocessors
+
+Use this section to change the [severity level](#severity-levels) of any rules in your extended configurations. Some rules may also receive [additional configurations](#additional-rule-options).
+
+* **type**: `array of objects`
+* **possible values**: `error`, `warn`, `off`
+
+Preprocessors run first during `lint` and `bundle`.
```
[the remark this answers is not in the document]
Comment (adamaltman) [reply docs/configuration/lint.md:165]: @RomanHotsiy do you know this?

[the remark this answers is not in the document]
Comment (RomanHotsiy) [reply docs/configuration/lint.md:165]: Yes, they do run.

https://github.com/Redocly/openapi-cli/blob/master/packages/core/src/bundle.ts#L78-L97

File: docs/configuration/lint.md [hunk 2]
```diff
@@ -42,45 +27,190 @@ lint:
...
+```    
+
+### Decorators
+
+Use this section to enable or disable decorators. They modify the definition in the bundling process after validation is complete.
+
+* **type**: `array of objects`
+* **possible values**: `error`, `warn`, `off`
+
+For `bundle` command: decorators run *after* linting.
```
[the remark this answers is not in the document]
Comment (RomanHotsiy) [reply docs/configuration/lint.md:215]: yes, if `-lint` is not set linting does not happen during bundle

File: docs/configuration/lint.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (RomanHotsiy) [reply docs/configuration/lint.md:165]: @adamaltman  ☝️

**Comment by bandantonio** [comment]
@adamaltman @RomanHotsiy Please proceed with another round of review

**Comment by RomanHotsiy** [comment]
@bandantonio something seems to be off:

image

**Pull request closed without merging**